### ISSUE TITLE
Add `pubspec.lock` to repository

### ISSUE THREAD

**Issue by mtrezza** [parse-community/Parse-SDK-Flutter#749]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Current Limitation
The `pubspec.lock` file is not checked in to this repository. This makes development more unstable and testing more volatile.

**Comment by fischerscode** [comment]
Since it won't be published to pub.dev (I think regardless of `.pubignore`, but I'm not sure about that.) it's just a development thing.
From that perspective: +1 for your argumentation
An alternative for CI: dump `pubspec.lock` or upload it as an artifact if a test fails.

> What we'd have to do manually is delete the lock file form time to time

`flutter pub upgrade --major-versions` might be our friend. We could even do some GitHub Actions magic.

**Comment by mtrezza** [comment]
Sounds good. I was surprised though that even popular dart packages do not have the lock file in their GitHub repo. So despite the arguments I am still hesitant to commit the lock file.

**Comment by mbfakourii** [comment]
@mtrezza 

Where did the issue go, should pubspec.lock be added?

**Comment by mtrezza** [comment]
This is still up for discussion.

**Comment by mtrezza** [comment]
Makes sense, how can *not* ignore it in GitHub, so that it's pushed to the repo, but then ignored during release?

**Comment by mbfakourii** [comment]
I think we should use ```.pubignore```, see [here](https://dart.dev/tools/pub/publishing#preparing-to-publish)

**Comment by mtrezza** [comment]
Sounds good to me.
